### PR TITLE
BHBC-1902: Add GeoServer

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to Jira tickets

https://quartech.atlassian.net/browse/BHBC-1902

### Added GeoServer based on the docker setup from (https://github.com/kartoza/docker-geoserver).
- Had to create a fork, as modifications were needed to make their original setup compatible with OpenShift
  - Fork is here: https://github.com/NickPhura/docker-geoserver

### Added `Containers/geoserver/...`
  - Contains OpenShift templates and a README explaining how to add and build them
    - This results in an instance of GeoServer running in OpenShift that can be accessed via a Route (with username/password)

### Added `geoserver/...` (removed for now as it was unused)
  - Contains the github action pipeline code to build and run GeoServer as part of our typical PR/Merge build/deployment pipeline.
    - This is not being used just yet in the git actions, as we don't have any code that leverages GeoServer yet, so no point building it everytime we do a PR.  Possible we might not want to anyways, and we may want to treat this like elastic search (where only 1 instance exists, and all PRs/dev make requests to it).

_On the fence about whether or not I should even include this in this PR, as its unclear if we will actually be deploying PR instances of GeoServer, or just treating it like ElasticSearch_

### Add geoserver to docker-compose
  - This won't build by default, you need to specifically call `make geoserver`.
    - Will be accessible at `localhost:8100`
    - relevant env vars are in `env.docker`

### The dev instance of GeoServer is accessible here: http://geoserver-dev-biohub-platform.apps.silver.devops.gov.bc.ca
- username: admin
- password: \<see geoserver secret in tools environment\>

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

**Review by anissa-agahchen — DISMISSED** [review]

File: Makefile [hunk 1]
```diff
@@ -9,9 +9,6 @@
 # Apply the contents of the .env to the terminal, so that the docker-compose file can use them in its builds
 export $(shell sed 's/=.*//' .env)
 
-.DEFAULT : help
-.PHONY : setup close clean build-backend run-backend build-web run-web database app api db-setup db-migrate db-rollback n8n-setup n8n-export clamav install test cypress lint lint-fix format format-fix help
```
Comment (curtisupshall) [inline Makefile:13]: Make doesn't care if recipes are "phony" or not?

File: database/README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 | Technology | Version | Website                              | Description          |
 | ---------- | ------- | ------------------------------------ | -------------------- |
-| node       | 10.x.x  | https://nodejs.org/en/               | JavaScript Runtime   |
+| node       | 14.x.x  | https://nodejs.org/en/               | JavaScript Runtime   |
 | npm        | 6.x.x   | https://www.npmjs.com/               | Node Package Manager |
 | PostgreSQL | 12.5    | https://www.postgresql.org/download/ | PSQL database        |
 | PostGIS    | 3       | https://postgis.net/                 | GIS (spatial) tools  |
```
Comment (curtisupshall) [inline database/README.md:10]: May as well label PostgreSQL as `14` and PostGIS as `3.1`?

File: api/README.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 | Technology | Version | Website                              | Description          |
 | ---------- | ------- | ------------------------------------ | -------------------- |
-| node       | 10.x.x  | https://nodejs.org/en/               | JavaScript Runtime   |
+| node       | 14.x.x  | https://nodejs.org/en/               | JavaScript Runtime   |
 | npm        | 6.x.x   | https://www.npmjs.com/               | Node Package Manager |
 | PostgreSQL | 9.6     | https://www.postgresql.org/download/ | PSQL database        |
```
Comment (curtisupshall) [inline api/README.md:9]: See other comment re: README version for Postges

File: geoserver/package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "npm": ">= 6.0.0"
+  },
+  "scripts": {
+    "lint": "eslint src/ --ext .js,.ts",
+    "lint-fix": "npm run lint -- --fix",
+    "format": "prettier --check \"./src/**/*.{js,jsx,ts,tsx,css,scss}\"",
+    "format-fix": "prettier --write \"./src/**/*.{js,jsx,ts,tsx,json,css,scss}\""
+  },
+  "dependencies": {
+    "typescript": "~4.1.6"
```
Comment (curtisupshall) [inline geoserver/package.json]: Should this be a dev dependency?

File: database/README.md (hunk 2 above)
Comment (NickPhura) [reply database/README.md:10]: as per our chat, reverting back to 12.5 for now.

File: Makefile (hunk 1 above)
Comment (NickPhura) [reply Makefile:13]: nothing changed when I took it out, and it was already missing things, so I just removed it

File: geoserver/package.json (hunk 4 above)
Comment (NickPhura) [reply geoserver/package.json]: This is inherited from other packages. We could test and see if its okay with moving, but I'm not confident either way off the top of my head.  Probably an issue for another ticket tho

**Review by KjartanE — DISMISSED** [review]
Looks good and is functioning!

**Review by KjartanE — APPROVED** [review]
🥇

**Pull request merged**